spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -175,7 +185,45 @@ public Statistics estimateStatistics() {
   protected Statistics estimateStatistics(Snapshot snapshot) {
     // its a fresh table, no data
     if (snapshot == null) {
-      return new Stats(0L, 0L);
+      return new Stats(0L, 0L, Collections.emptyMap());
+    }
+
+    boolean cboEnabled =
+        Boolean.parseBoolean(spark.conf().get(SQLConf.CBO_ENABLED().key(), "false"));
+    Map<NamedReference, ColumnStatistics> colStatsMap = null;
```

Comment:
Thinking back to this change, may be cleaner to just set this to empty map and the reassign to Maps.newHashMap on 195 as is. 

colStatsMap = Collections.emptyMap
if (report & cbo) {
   colStatsMap = Maps.newHashMap
}